src/endpoint/s3/ops/s3_put_bucket_policy.js

```diff
@@ -23,10 +23,38 @@ async function put_bucket_policy(req) {
         }
         if (statement.action) statement.action = _.flatten([statement.action]);
         if (statement.resource) statement.resource = _.flatten([statement.resource]);
+        if (statement.condition) statement.condition = _parse_condition(statement.condition);
     }
     await req.object_sdk.put_bucket_policy({ name: req.params.bucket, policy });
 }
 
+function _parse_condition(conditions) {
+    const valid_conditions = {
+        'stringequals': 'string_equals',
+        'stringnotequals': 'string_not_equals',
+        'null': 'null'
+    };
+    const condition_key_map = {
+        's3:x-amz-server-side-encryption': 'server_side_encryption'
+    };
+    const parsed_condition = {};
+    for (const [condition, obj] of Object.entries(conditions)) {
+        if (!valid_conditions[condition]) {
+            console.error('put_bucket_policy: invalid condition', condition);
```

Comment:
This Error is not indicating that we are not supporting the condition, for example if we get a condition that is valid such as `StringEqualsIgnoreCase` but we are not supporting it we should prompt the error that indicate that it is not supported.
also we can use dbg and not console
```suggestion
            dbg.error('put_bucket_policy: not supported condition', condition);
```